### PR TITLE
Document LuigiElement feature on docu website

### PR DESCRIPTION
**Pull request by amilewskaa**
**Description**

Changes proposed in this pull request:

- Document LuigiElement feature on docu website

**Related issue(s)**

#3881

### CODE REVIEW

File: scripts/package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "docu:container:generate:compound-container-api": "documentation readme ../container/typings/LuigiCompoundContainer.svelte.d.ts --parse-extension ts -f md --readme-file=../docs/luigi-compound-container-api.md --section=\"API Reference\" --markdown-toc=false --quiet --github false",
     "docu:client": "npm run docu:client:validate && npm run docu:client:generate:section",
     "docu:client:generate:section": "documentation readme ../client/src/luigi-client.js ../client/luigi-client-wc-docu-mixin.js -f md --readme-file=../docs/luigi-client-api.md --section=\"API Reference\" --markdown-toc=false --quiet --github false",
+    "docu:element": "documentation readme ../client/src/luigi-element.js -f md --readme-file=../docs/luigi-element-api.md --section=\"API Reference\" --markdown-toc=false --quiet --github false",
```
Comment (walmazacn) [inline scripts/package.json:14]: You should also add this command to `docu` command in line 5 to be sure output file is built automatically while running documentation task

**Review by walmazacn — CHANGES_REQUESTED** [review]

**Review by walmazacn — APPROVED** [review]
Looks good 👍

File: docs/luigi-element-api.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  afterInit(ctx) {
+    console.log('MyComponent initialized with context:', ctx);
+  }
+
+  onContextUpdate(ctx) {
+    console.log('Context updated:', ctx);
+  }
+}
+
+customElements.define('my-component', MyComponent);
```
Comment (JohannesDoberer) [inline docs/luigi-element-api.md:40]:
```suggestion

```
Please remove this line. We recommend that web component developers do not define their own tag name. Luigi registers the web component with a generated unique tagname. The reason we do not recommend this is the possibility of multiple web components having the same tagname, which can lead to conflicts.

**Comment by JohannesDoberer** [comment]
I checked the deployed preview version and I see that the marker for functions is missing, e.g.
Bildschirmfoto 2024-11-12 um 10 53 29

File: docs/luigi-element-api.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      "categoryPosition": 6,
+      "position": 5
+    }
+  }
+}
+meta -->
+
+# Luigi Element
+
+This document outlines the parameters provided by the Luigi Luigi Element. The `LuigiElement` class is a base class for building web component micro frontends within the Luigi framework. It provides essential functionality for rendering and managing the lifecycle of web components.
```
Comment (hardl) [inline docs/luigi-element-api.md:19]: Luigi Luigi Element sounds a bit strange ;) 
I'd probably simply remove the whole first sentence

**Review by hardl — CHANGES_REQUESTED** [review]

File: docs/luigi-element-api.md (hunk 3 above)
Comment (amilewskaa) [reply docs/luigi-element-api.md:19]: I followed the pattern of other API pages - they all start with such sentence 😄 

But ofc I can remove it if you want 😸

**Review by hardl — APPROVED** [review]

**Pull request merged**